agent/php_curl_md.c

```diff
@@ -42,76 +35,142 @@ static void nr_php_curl_md_destroy(nr_php_curl_md_t* metadata) {
   })
 #endif
 
-#define ensure_curl_metadata_hashmap()                                       \
-  if (!NRTXNGLOBAL(curl_metadata)) {                                         \
-    NRTXNGLOBAL(curl_metadata)                                               \
-        = nr_hashmap_create((nr_hashmap_dtor_func_t)nr_php_curl_md_destroy); \
+static void nr_php_curl_multi_md_destroy(
+    nr_php_curl_multi_md_t* multi_metadata) {
+  nr_vector_deinit(&multi_metadata->curl_handles);
+  nr_free(multi_metadata->async_context);
+  nr_free(multi_metadata);
+}
+
+static void nr_php_curl_md_destroy(nr_php_curl_md_t* metadata) {
+  nr_php_zval_free(&metadata->outbound_headers);
+  nr_free(metadata->method);
+  nr_free(metadata->response_header);
+  nr_free(metadata);
+}
+
+static void curl_handle_vector_dtor(void* element, void* userdata NRUNUSED) {
+  zval* handle = (zval*)element;
+
+  if (!handle) {
+    return;
   }
 
-#define ensure_curl_multi_metadata_hashmap()                   \
```

Comment:
I couldn't find a reason why several of these macros were used instead of functions. For easier readability- as well as less code duplication when adding the ifdefs to fix the issue- I changed them to functions. I also moved some functions around so that definitions were in the correct order.